sycl/test-e2e/LLVMIntrinsicLowering/bitreverse.cpp

```diff
@@ -0,0 +1,189 @@
+    return (a << 16) | (a >> 16);
+}
+template <typename UINT32>
+void do_bitreverse_test32() {
+  for (uint32_t u=0 ; u<(0xFF<<24); u+=0xABCD13) {
+    UINT32 data=u;
+    UINT32 ref[1];
+    UINT32 opt[1];
+
+    ref[0] = reference_reverse32(data);
+    opt[0] = reverse32(data);    
+    if (memcmp(ref,opt,sizeof(UINT32))) return;
+  }
+  ext::oneapi::experimental::printf("Passed bitreverse 32 %2d\n",sizeof(UINT32));  
+}
+
+////////////////////////////////////////////////////////////////////////////////////
+// 64-bit
+////////////////////////////////////////////////////////////////////////////////////
+template <typename UINT64>
+__attribute__((optnone, noinline)) UINT64 reference_reverse64(UINT64 a) {
+    a = ((0x5555555555555555UL & a) << 1) | (0x5555555555555555UL & (a >> 1));
+    a = ((0x3333333333333333UL & a) << 2) | (0x3333333333333333UL & (a >> 2));
+    a = ((0x0F0F0F0F0F0F0F0FUL & a) << 4) | (0x0F0F0F0F0F0F0F0FUL & (a >> 4));
+    a = ((0x00FF00FF00FF00FFUL & a) << 8) | (0x00FF00FF00FF00FFUL & (a >> 8));
+    a = ((0x0000FFFF0000FFFFUL & a) <<16) | (0x0000FFFF0000FFFFUL & (a >>16));    
+    return (a << 32) | (a >> 32);
+}
+template <typename UINT64>
+__attribute__((noinline)) UINT64 reverse64(UINT64 a) {
+    a = ((0x5555555555555555UL & a) << 1) | (0x5555555555555555UL & (a >> 1));
+    a = ((0x3333333333333333UL & a) << 2) | (0x3333333333333333UL & (a >> 2));
+    a = ((0x0F0F0F0F0F0F0F0FUL & a) << 4) | (0x0F0F0F0F0F0F0F0FUL & (a >> 4));
+    a = ((0x00FF00FF00FF00FFUL & a) << 8) | (0x00FF00FF00FF00FFUL & (a >> 8));
+    a = ((0x0000FFFF0000FFFFUL & a) <<16) | (0x0000FFFF0000FFFFUL & (a >>16));    
+    return (a << 32) | (a >> 32);
+}
+template <typename UINT64>
+void do_bitreverse_test64() {
+  for (uint64_t u=0 ; u<(0xFFUL<<56); u+=0xABCDABCDABCD13UL) {
+    UINT64 data=u;
+    UINT64 ref[1];
+    UINT64 opt[1];
+
+    ref[0] = reference_reverse64(data);
+    opt[0] = reverse64(data);    
+    if (memcmp(ref,opt,sizeof(UINT64))) return;
+  }
+  ext::oneapi::experimental::printf("Passed bitreverse 64 %2d\n",sizeof(UINT64));
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+template <class T> class BitreverseTest;
+
+template <typename UINT8, typename UINT16, typename UINT32, typename UINT64>
+void testTypes() {
+  queue q;
+
+  q.submit([](handler &cgh) { cgh.single_task<BitreverseTest<UINT8>> ([]() { do_bitreverse_test8 <UINT8> (); }); }); q.wait();
+  q.submit([](handler &cgh) { cgh.single_task<BitreverseTest<UINT16>>([]() { do_bitreverse_test16<UINT16>(); }); }); q.wait();
+  q.submit([](handler &cgh) { cgh.single_task<BitreverseTest<UINT32>>([]() { do_bitreverse_test32<UINT32>(); }); }); q.wait();
+  q.submit([](handler &cgh) { cgh.single_task<BitreverseTest<UINT64>>([]() { do_bitreverse_test64<UINT64>(); }); }); q.wait();  
+}
+
+typedef  uint8_t  uint8_t2  __attribute__((ext_vector_type(2)));
```

Comment:
nit: we can use the `using` keyword instead of `typedef`

Reply:
Updated to use using keyword instead of typedef.